packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -282,9 +283,28 @@ export class VueElement extends BaseClass {
 
     const asyncDef = (this._def as ComponentOptions).__asyncLoader
     if (asyncDef) {
-      asyncDef().then(def => resolve(def, true))
+      asyncDef().then(def => {
+        resolve(def, true)
+        if (this._childResolve.length > 0) {
+          this._childResolve.forEach((childResolveFn: () => void) => {
+            childResolveFn()
+          })
+        }
+      })
     } else {
-      resolve(this._def)
+      // When parentNode is a custom element rendered by an asynchronous component,
+      // it is collected and waits for the asynchronous
+      // completion before executing the resolve function.
+      const parentNode = this.parentNode as typeof this
```

Comment:
It seems not right, `parentNode` may not be a ce component
```html
<my-el-async-nested-ce>
  <div>
    <slotted-child></slotted-child>
  </div>
</my-el-async-nested-ce>
```

Reply:
What about this idea? Recursively search up the async custom element `parentCE`. If `parentCE` is found, process the `resolve` related logic of the child component in its then function, and pass `parentCE` to `resolve` to avoid recursing `parent` to process `porvides`